sonic_platform_base/sonic_xcvr/api/public/cmis.py

```diff
@@ -1290,6 +1310,8 @@ def reset_module(self, reset = False):
         Return True if no action.
         '''
         if reset:
+            # Clear cached values when module is reset (module removal event)
```

Comment:
@jianyuewu there is no guarantee platforms has to call this reset_module() API on OIR

Reply:
@prgeor OK, is there some suitable place to call clear_cache()? Thank you😊 I checked the code, didn't find suitable place yet, previously thought maybe put in get_xcvr_api() when it is not present, but might cost some time every call in get_xcvr_api(). Is that acceptable?

Reply:
Calling `get_present()` in `get_xcvr_api()` cannot guarantee the correctness. It could be the case that `get_xcvr_api()` is called after a "plug out" ---- "plug in" sequence. In that case, `get_present` returns True but cable was replaced. Some options from my side:

1. Make the cache configurable. The cache is disabled by default. We enable it on NVIDIA platform. Other vendors may choose enable/disable it by themselves.
2. Calling `clear_cache` in `chassis.get_change_event` when module is removed. Currently, only xcvrd calls `get_change_event`, so it means that other process may not detect the module plug out event and cannot clear the cache.

Reply:
Code updated😊